esp-hal/src/dma/pdma.rs

```diff
@@ -857,23 +749,24 @@ impl InterruptAccess<DmaRxInterrupt> for AnyI2sDmaRxChannel {
     }
 }
 
-macro_rules! ImplI2sChannel {
-    ($num: literal) => {
+macro_rules! ImplPdmaChannel {
+    ($peri:ident, $num:literal, $int:ident) => {
         paste::paste! {
-            #[doc = concat!("DMA channel suitable for I2S", $num)]
-            pub struct [<I2s $num DmaChannel>] {}
+            #[doc = concat!("DMA channel suitable for ", stringify!([< $peri:upper >]), $num)]
+            #[non_exhaustive]
```

Comment:
I guess this bug has lived long enough, no more safely constructed I2S DMA channels 🙃

Reply:
Not anymore, these are the channel "peripherals".